app/src/main/java/com/blink22/android/mvpandroid/todos/TodosPresenter.java

```diff
@@ -0,0 +1,97 @@
+package com.blink22.android.mvpandroid.todos;
+
+import android.arch.lifecycle.Lifecycle;
+import android.arch.lifecycle.LifecycleObserver;
+import android.arch.lifecycle.OnLifecycleEvent;
+
+import com.blink22.android.mvpandroid.models.Todo;
+import com.blink22.android.mvpandroid.network.TodosSubscriber;
+
+import java.lang.reflect.Array;
+import java.util.ArrayList;
+
+import io.reactivex.disposables.CompositeDisposable;
+import io.reactivex.observers.DisposableObserver;
+import rx.Subscription;
+import rx.subscriptions.CompositeSubscription;
+
+/**
+ * Created by ahmedghazy on 7/25/18.
+ */
+
+public class TodosPresenter implements TodosContract.Presenter {
+    private static final String TAG = TodosPresenter.class.getSimpleName();
+    private static TodosPresenter sTodosPresenter;
+
+    private final TodosSubscriber mTodosSubscriber;
+    private final TodosContract.View mView;
+    private CompositeDisposable mCompositeDisposable;
+    private ArrayList<Todo> mTodos;
+
+    public static TodosPresenter getInstance(TodosSubscriber todosSubscriber, TodosContract.View view) {
+        return sTodosPresenter == null ? new TodosPresenter(todosSubscriber, view) : sTodosPresenter;
+    }
+
+    private TodosPresenter(TodosSubscriber todosSubscriber, TodosContract.View view) {
+
+        this.mTodosSubscriber = todosSubscriber;
+        this.mView = view;
+
+    }
+
+    @Override
+    @OnLifecycleEvent(Lifecycle.Event.ON_CREATE) public void start() {
+        this.mCompositeDisposable = new CompositeDisposable();
+    }
+
+    @Override
+    @OnLifecycleEvent(Lifecycle.Event.ON_DESTROY) public void stop() {
```

Comment:
@AFGhazy I suggest putting the annotations in a separate line abovew the method signature as you did in `NewTodoPresenter` Check [this](https://github.com/ribot/android-guidelines/blob/master/project_and_code_guidelines.md#2262-annotations-style)